packages/flutter_tools/lib/src/reporting/crash_reporting.dart

```diff
@@ -26,15 +26,61 @@ const String _kStackTraceFileField = 'DartError';
 /// it must be supplied in the request.
 const String _kStackTraceFilename = 'stacktrace_file';
 
+class CrashDetails {
+  CrashDetails({
+    @required this.command,
+    @required this.error,
+    @required this.stackTrace,
+    @required this.doctorText,
+  });
+
+  String command;
+  dynamic error;
+  StackTrace stackTrace;
+  String doctorText;
+}
+
+/// Reports information about the crash to the user.
+class CrashReporter {
+  CrashReporter({
+    @required this.logger,
+    this.gitHubTemplateCreator,
+  });
+
+  final GitHubTemplateCreator gitHubTemplateCreator;
+  final Logger logger;
+
+  /// Prints instructions for filing a bug about the crash.
+  Future<void> informUser(CrashDetails details, File crashFile) async {
+    logger.printError('A crash report has been written to ${crashFile.path}.');
+    logger.printStatus('This crash may already be reported. Check GitHub for similar crashes.', emphasis: true);
+
+    final String similarIssuesURL = GitHubTemplateCreator.toolCrashSimilarIssuesURL('${details.error}');
```

Comment:
details.error.toString()

Reply:
Is there a preference for using `.toString()`?  I generally prefer string interpolation since it guarantees a string even for null inputs.

Reply:
IIRC from my VM team days, interpolation will be a bit slower. This code is not performance critical, so the overriding concern here would be following a consistent style. Also, I think it is not possible to throw `null`, so `error` won't  ever be `null`. We'll be able to enforce that to this point with non-nullability in the future.

Reply:
Okay, I'll fix it `.toString()`.